src/Symfony/Component/Messenger/Event/WorkerMessageHandlingEvent.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Event;
+
+/**
+ * Dispatched when a message was received from a transport but before sent to the bus.
+ *
+ * The event name is the class name.
+ *
+ * @experimental in 4.3
+ */
+class WorkerMessageHandlingEvent extends AbstractWorkerMessageEvent
```

Comment:
I'd name it `WorkerMessageReceivedEvent` or `WorkerWillHandleMessageEvent`, otherwise it's not clear enough IMO.